### PR TITLE
Kga to object

### PR DESCRIPTION
**Pull request by kevinpapst**
Changes proposed in this pull request:
- Large refactoring of the $kga to an ArrayObject, allowing backward compatibility with the array usage
- The new object structure allows for OO programming
- Moved Authenticator to Registry (fixes kimai/roadmap#3)
- Rewritten Debug extension to not-preload data, but use JS always / allow for easier debugging the kga object / removed reload button / removed unused images
- added Kimai_Translation_Service and Kimai_Translation_Data, preventing multiple unnecessary data loads
- rewritten config loading, utilizing the new Config ArrayObjects (preventing a lot of duplicated database requests)
- removed vars.php - logic went into new Config objects
- worked on some translations
- code formattings on the fly
- fixed editLimit check in editing and creating expenses
- ... and some more tweaks, see commits

Reason for this pull request:
- Future improvements of Kimai and allowing more OO programming style

### CODE REVIEW

File: extensions/ext_debug/processor.php [hunk 1]
```diff
@@ -83,35 +81,71 @@
...
+                if (is_array($v)) {
+                    foreach($v as $k2 => $v2) {
+                        if (isset($output[$k]) && isset($output[$k][$k2])) {
+                            $output[$k][$k2] = $v2;
+                        }
+                    }
+                } else {
+                    $output[$k] = $v;
+                }
+            }
```
Comment (simonschaufi) [inline extensions/ext_debug/processor.php:145]: how about extracting this into a function?

**Comment by kevinpapst** [comment]
I will rebase this pull request, once we merged the larger PRs for 1.1!

**Comment by kevinpapst** [comment]
@simonschaufi Please test this branch. I updated it to develop, but its quite some time ago that I created that branch and the merge was a little painful ;-)

**Comment by simonschaufi** [comment]
There is still something to do:

```
@C:\xampp\htdocs\libraries\Kimai\Format.php line 72
[27.08.2016 15:15:00] E_NOTICE Undefined index: show_gabBreaks @C:\xampp\htdocs\extensions\ki_timesheets\templates\scripts\timeSheet.php line 57
[27.08.2016 15:15:00] E_NOTICE Undefined index: decimalSeparator @C:\xampp\htdocs\extensions\ki_timesheets\templates\scripts\timeSheet.php line 146
[27.08.2016 15:15:07] E_NOTICE Undefined index: roundMinutes @C:\xampp\htdocs\extensions\ki_timesheets\templates\scripts\floaters\add_edit_timeSheetEntry.php line 44
[27.08.2016 15:15:07] E_NOTICE Undefined index: roundSeconds @C:\xampp\htdocs\extensions\ki_timesheets\templates\scripts\floaters\add_edit_timeSheetEntry.php line 45
[27.08.2016 15:15:07] E_NOTICE Undefined index: roundTimesheetEntries @C:\xampp\htdocs\extensions\ki_timesheets\templates\scripts\floaters\add_edit_timeSheetEntry.php line 46
[27.08.2016 15:15:07] KSPI axAction (admin): reload_activities_options
[27.08.2016 15:15:21] KSPI axAction (admin): add_edit_timeSheetEntry
[27.08.2016 15:15:21] E_NOTICE Undefined index: roundMinutes @C:\xampp\htdocs\extensions\ki_timesheets\templates\scripts\floaters\add_edit_timeSheetEntry.php line 44
[27.08.2016 15:15:21] E_NOTICE Undefined index: roundSeconds @C:\xampp\htdocs\extensions\ki_timesheets\templates\scripts\floaters\add_edit_timeSheetEntry.php line 45
[27.08.2016 15:15:21] E_NOTICE Undefined index: roundTimesheetEntries @C:\xampp\htdocs\extensions\ki_timesheets\templates\scripts\floaters\add_edit_timeSheetEntry.php line 46
```
- I noticed that the "Lohn" column is not correct. Same in export.
  Example: 005
- In debug tab: selecting "user" i can see the whole object without "xxx"-ed values.
- In Invoice tab the timespan is totally broken:
  from is empty and to is "%27.%8.%Y"

**Comment by kevinpapst** [comment]
Are you sure that you used the correct version of this branch? I do not have any of these problems.

**Comment by simonschaufi** [comment]
I found the problem. Mysql returns an error in my case:

```
SELECT * FROM `kimai_configuration` WHERE option NOT IN ('version', 'revision')
```

Try to execute that in phpmyadmin and you will get:

```
#1064 - You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'option NOT IN ('version', 'revision')
LIMIT 0, 25' at line 1
```

**Comment by simonschaufi** [comment]
please apply: 
[unnamed1.patch.txt](https://github.com/kimai/kimai/files/441360/unnamed1.patch.txt)

**Comment by kevinpapst** [comment]
thanks 👍  applied

**Comment by kevinpapst** [comment]
hahaha, da hatte es aber jemand eilig^^

**Pull request merged**